lib/matplotlib/projections/geo.py

```diff
@@ -269,24 +269,26 @@ def __init__(self, resolution):
             self._resolution = resolution
 
         def transform_non_affine(self, ll):
-            longitude = ll[:, 0:1]
-            latitude  = ll[:, 1:2]
+            longitude = ll[:, 0]
+            latitude = ll[:, 1]
 
             # Pre-compute some values
             half_long = longitude / 2.0
             cos_latitude = np.cos(latitude)
 
             alpha = np.arccos(cos_latitude * np.cos(half_long))
             # Mask this array or we'll get divide-by-zero errors
-            alpha = ma.masked_where(alpha == 0.0, alpha)
-            # The numerators also need to be masked so that masked
-            # division will be invoked.
+            mask = alpha != 0.0
+            alpha = alpha[mask]
```

Comment:
A simpler alternative would be to use the general method that numpy uses in its sinc:
```
    alpha[alpha == 0] = 1e-20
    sinc_alpha = sin(alpha)/alpha
```
But it looks like there is more to it than just getting sinc(0) = 1.  See below.

Reply:
The singular point is just the center point, (x, y) =  (lon, lat) = (0, 0).  So it looks like the modification suggested above would work fine, eliminating all of the subsequent indexing, simplifying and speeding up the code.

Reply:
Working through the same thing below, `alpha = 0` from `arccos` comes from input of 1 from `cos_latitude * np.cos(half_long)`. Using the same limits, that can only happen if `longitude = latitude = 0`. To ensure that the result is 0 with sinc(0) = 1, the numerator of both calculations needs to be 0. Fortunately, we have `sin(longitude|latitude)` in both which nets 0 there too.

I think this will work out just as well.